Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+        {
+            if (bytes == null)
+                throw new ArgumentNullException ("bytes");
+
+            int blen = bytes.Length;
+            if (blen == 0)
+                return new byte [0];
+
+            if (offset < 0 || offset >= blen)
+                throw new ArgumentOutOfRangeException("offset");
+
+            if (count < 0 || count > blen - offset)
+                throw new ArgumentOutOfRangeException("count");
+
+            MemoryStream result = new MemoryStream (count);
+            int end = offset + count;
+            for (int i = offset; i < end; i++)
+                UrlEncodeChar ((char)bytes [i], result, false);
+
+            return result.ToArray();
+        }
+
+        internal static string HtmlEncode (string s)
+        {
+            if (s == null)
+                return null;
+
+            if (s.Length == 0)
+                return String.Empty;
+
+            bool needEncode = false;
+            for (int i = 0; i < s.Length; i++) {
+                char c = s [i];
+                if (c == '&' || c == '"' || c == '<' || c == '>' || c > 159
+                    || c == '\''
+                ) {
+                    needEncode = true;
+                    break;
+                }
+            }
+
+            if (!needEncode)
+                return s;
+
+            StringBuilder output = new StringBuilder ();
+            int len = s.Length;
+
+            for (int i = 0; i < len; i++) {
+                char ch = s [i];
+                switch (ch) {
+                    case '&' :
+                        output.Append ("&amp;");
+                        break;
+                    case '>' :
+                        output.Append ("&gt;");
+                        break;
+                    case '<' :
+                        output.Append ("&lt;");
+                        break;
+                    case '"' :
+                        output.Append ("&quot;");
+                        break;
+                    case '\'':
+                        output.Append ("&#39;");
+                        break;
+                    case '\uff1c':
+                        output.Append ("&#65308;");
+                        break;
+
+                    case '\uff1e':
+                        output.Append ("&#65310;");
+                        break;
+
+                    default:
+                        if (ch > 159 && ch < 256) {
+                            output.Append ("&#");
+                            output.Append (((int) ch).ToString (Helpers.InvariantCulture));
+                            output.Append (";");
+                        } else
+                            output.Append (ch);
+                        break;
+                }
+            }
+
+            return output.ToString ();
+        }
+
+        internal static string HtmlAttributeEncode (string s)
+        {
+            if (String.IsNullOrEmpty (s))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)